src/core/ComplexFilter/index.tsx

```diff
@@ -88,25 +89,29 @@ export default function ComplexFilter<
         </div>
       </Wrapper>
       <PopperComponent open={open} anchorEl={anchorEl}>
-        <MenuSelect
-          open
-          search={search}
-          onClose={handleClose}
-          multiple={multiple as Multiple}
-          PaperComponent={PaperComponent}
-          value={
-            (multiple ? pendingValue : value) as MUIValue<
-              DefaultMenuSelectOption,
-              Multiple,
-              undefined,
-              undefined
-            >
-          }
-          onChange={handleChange}
-          disableCloseOnSelect={multiple}
-          options={options}
-          {...MenuSelectProps}
-        />
+        <ClickAwayListener onClickAway={handleClose}>
```

Comment:
amazing

Reply:
Thank you 😄 ! Actually found this from v5's Autocomplete Github Picker [example](https://mui.com/components/autocomplete/#githubs-picker), which their [v4 example](https://v4.mui.com/components/autocomplete/#githubs-picker) doesn't use and has the bug of clicking the click target won't dismiss the dropdown overlay 💡 

So using `<ClickAwayListener />` here is a bonus fix for that particular bug!

![demo](https://user-images.githubusercontent.com/6309723/146614519-a08bb96e-ae74-4bc1-a8b1-233f92439699.gif)